service/lock/mutex_lock.go

```diff
@@ -0,0 +1,46 @@
+package lock
+
+import (
+	"context"
+	"sync"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+type MutexLockConfig struct {
+	Logger micrologger.Logger
+}
+
+// MutexLock implements Interface using sync.Mutex. For now we use a shared
+// instance of *MutexLock for all IPAM related activity of network packages in
+// the legacy controllers and ipam resources in the clusterapi controllers.
+type MutexLock struct {
+	logger micrologger.Logger
+
+	mutex sync.Mutex
```

Comment:
I think when sync.Mutex is used we should use it directly. When we need something more sophisticated we should use it instead. I don't see a reason for interface here. We are not going to change locking for tests. This only obscures the code IMO.

Reply:
We use the in process mutex locker now. We will need an inter process locker when the operators are flattened and we run multiple ones. An interface makes sense to transition later.

Reply:
We will replace old implementation with the new one. I still don't see the need for the interface. We can sort that later though.

Reply:
But yeah I understand what you want to achieve here.